### PR TITLE
feat: signup to supabase auth

### PR DESCRIPTION
**Pull request by ValMgr**
- [x] basic form
- [x] request supabase to create user
- [x] handle password and mail validators 
- [x] handle errors
- [x] handle successfull sign-up
- [x] check if email exists in ECV students list
- [x] create layout around form

### CODE REVIEW

File: src/app.d.ts [hunk 1]
```diff
@@ -2,8 +2,11 @@
 // for information about these interfaces
 // and what to do when importing types
 declare namespace App {
-	// interface Locals {}
-	// interface PageData {}
-	// interface Error {}
-	// interface Platform {}
+	interface Supabase {
+		Database: import('./DatabaseDefinitions').Database;
```
Comment (iOiurson) [inline src/app.d.ts:6]: I don't know what this import is. I don't see it anywhere.

Also, i'm not sure these Supabase typings should live in this file.

Generally, this `app.d.ts` file should rather be for SvelteKit specific type definitions.

File: src/routes/+page.svelte [hunk 2]
```diff
@@ -1,18 +1,9 @@
...
+	import Link from '$lib/components/Link.svelte';
 	import Icon from '$lib/components/Icon.svelte';
 
-	let search = '';
-	let value = '';
-	let number = 0;
-	let bool_1 = false;
-	let bool_2 = false;
-	let radio_value = '';
+	import { goto } from '$app/navigation';
```
Comment (iOiurson) [inline src/routes/+page.svelte:6]: Just a detail, but i like my imports to have a specific order:
- Svelte stuff
- SvelteKit stuff
- Types
- Helpers
- General components
- Page specific components

This is just a proposal for this project, we can discuss it in class.

File: src/routes/+page.svelte [hunk 3]
```diff
@@ -23,26 +14,5 @@
...
-	<Checkbox name="checkbox_1" label={'Checkbox'} bind:value={bool_1} />
-
-	<Checkbox name="checkbox_2" reverse label={'Checkbox reversed'} bind:value={bool_2} />
-
-	<p>Buttons</p>
-
-	<Button />
-	<Button type="rounded" />
-	<Button type="square" />
+	<Button on:click={() => goto('/inscription', { replaceState: true })}>S'enregister</Button>
```
Comment (iOiurson) [inline src/routes/+page.svelte:17]: I believe this should be a `<a>` or a `<Link>`

**Review by iOiurson — CHANGES_REQUESTED** [review]

File: src/lib/components/forms/Input.svelte [hunk 4]
```diff
@@ -14,7 +14,7 @@
 <div class="group" class:onError={error}>
 	{#if !noLabel}
 		<label for={name}>
-			<slot>Label*</slot>
```
Comment (clemcode) [inline src/lib/components/forms/Input.svelte:17]: Required fields should be specified via HTML (https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/required) and not hard-coded as string.

File: src/routes/inscription/+page.server.ts [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+function isValidPassword(password: string): boolean {
+	const regex = /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{9,}$/;
+	return regex.test(password);
+}
+
+function isEmailValid(email: string): boolean {
+	const emailRegex = /^[^@]+@mail-ecv\.fr$/;
+	return emailRegex.test(email);
+}
```
Comment (clemcode) [inline src/routes/inscription/+page.server.ts:68]: These functions will be useful for other features, maybe we can extract them in a utils file to make them simpler to discover / reuse?

**Review by clemcode — CHANGES_REQUESTED** [review]

**Pull request closed without merging**